components/camel-optaplanner/src/main/java/org/apache/camel/component/optaplanner/OptaPlannerProducer.java

```diff
@@ -59,12 +63,38 @@ protected void doStop() throws Exception {
         super.doStop();
     }
 
-    @SuppressWarnings("unchecked")
     @Override
-    public synchronized void process(Exchange exchange) throws Exception {
-        final Object body = exchange.getIn().getMandatoryBody();
-        final String solverId = getSolverId(exchange);
+    public boolean process(Exchange exchange, AsyncCallback callback) {
```

Comment:
The callback must ALWAYS be called, so here is a few problems.

There is no return false for the async processing. And in case of an exception in try .. catch, you need to call callback.done(true) also (for example if mandatory body throws exception), which can be done in a finally block, or just above return true etc.